### DISCUSSION TITLE
Configuring different schedules for terraform modules according to registry

### DISCUSSION THREAD

**Discussion by jgrumboe** [renovatebot/renovate#8714] [Request Help]
Hi everyone,

I would like to configure renovate to give me PR asap for our own modules coming from our Terraform Cloud registry and other modules should be updated on a monthly basis, to reduce noise.

I tried the following:

```
  "packageRules": [
    {
      "packagePatterns": ["^app.terraform.io"],
      "schedule": "at any time"
    },
    {
      "managers": ["terraform"],
      "extends": ["schedule:monthly"]
    }
  ]
```

But the result is not the intended one, everything will be scheduled monthly, with no just-in-time updates for our own modules.
Could you help me?
How should the correct config look like?
Is renovate-config a first-match or last-match win?

Best regards,
Johannes

**Comment by rarkins** [comment]
Renovate config matches everything, so that probably means "last match win" according to your question. Try reversing the rules you have above.

**Reply by jgrumboe** [reply]
Thanks, @rarkins, for your response.
Reversing the order didn't solve it completely, BUT I managed to get it working by replacing the packagePattern with sourceUrlPrefix.

So my working configuration now is this one:
```
  "packageRules": [
    {
      "managers": ["terraform"],
      "extends": ["schedule:monthly"]
    },
    {
      "sourceUrlPrefixes": ["https://github.com/ourorganisation/"],
      "schedule": "at any time"
    }
  ]
```
Don't know what I've got wrong with my packagepattern but sourceurlprefix is also fine for me.

**Reply by rarkins** [reply]
If you look in the debug log for "packageFiles with updates" you can see what the "depName" is for each dependency. Perhaps it doesn't start with app.terraform.io like you think it does?

**Reply by jgrumboe** [reply]
Thanks for pointing me to depName, got it ... upper/lowercase problem 🤦
